cmd/kubeadm/app/util/config/initconfiguration_test.go

```diff
@@ -42,6 +42,7 @@ const (
 	masterInternalYAMLNonLinux  = "testdata/conversion/master/internal_non_linux.yaml"
 	masterIncompleteYAML        = "testdata/defaulting/master/incomplete.yaml"
 	masterDefaultedYAML         = "testdata/defaulting/master/defaulted.yaml"
+	masterDefaultedYAMLDarwin   = "testdata/defaulting/master/defaulted_darwin.yaml"
```

Comment:
@bart0sh 
this is the commit that added the change:
https://github.com/kubernetes/kubernetes/commit/c06ca11b5dc637fe3df3fb2b50a1980e1237654c#diff-a1fb776151491a03f106cf3d74660259

please try just adding these fields to all defaulted* instead of adding a new file:
```
winkernel:
  enableDSR: false
  networkName: ""
  sourceVip: ""
```

https://github.com/kubernetes/kubernetes/blob/a2a5bd03fdae6ac45509560157f4c24d1c001e86/staging/src/k8s.io/kube-proxy/config/v1alpha1/types.go#L154

^ the `winkernel` field is always present.

also please change this instance to lowercase `winkernel`:
https://github.com/kubernetes/kubernetes/commit/c06ca11b5dc637fe3df3fb2b50a1980e1237654c#diff-59f935b7bbcb24940f16ea5b3f9f9276R93

Reply:
New file removes only one field - nodefs.inodesFree:
```
$ diff -u cmd/kubeadm/app/util/config/testdata/defaulting/master/defaulted.yaml cmd/kubeadm/app/util/config/testdata/defaulting/master/defaulted_darwin.yaml
--- cmd/kubeadm/app/util/config/testdata/defaulting/master/defaulted.yaml	2019-02-22 16:37:45.000000000 +0100
+++ cmd/kubeadm/app/util/config/testdata/defaulting/master/defaulted_darwin.yaml	2019-02-23 12:16:57.000000000 +0100
@@ -119,7 +119,6 @@
   imagefs.available: 15%
   memory.available: 100Mi
   nodefs.available: 10%
-  nodefs.inodesFree: 5%
 evictionPressureTransitionPeriod: 5m0s
 failSwapOn: true
 fileCheckFrequency: 20s
```

winkernel only exists on Windows. That's why I changed the code to use the same files as on Linux with only one exception: masterDefaultedYAMLDarwin.

Reply:
we can remove the `defaulted_darwin.yaml` file.

the linux defaults are here:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/config/v1beta1/defaults_linux.go

the non-linux defaults:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/config/v1beta1/defaults_others.go

our `defaulted_non_linux.yaml` file complies with that:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/util/config/testdata/defaulting/master/defaulted_non_linux.yaml#L114-L117

Reply:
> we can remove the `defaulted_darwin.yaml` file.

I believe we can't:
```
$ diff -u cmd/kubeadm/app/util/config/testdata//defaulting/master/defaulted_non_linux.yaml cmd/kubeadm/app/util/config/testdata//defaulting/master/defaulted_darwin.yaml 
--- cmd/kubeadm/app/util/config/testdata//defaulting/master/defaulted_non_linux.yaml	2019-02-19 00:14:12.000000000 +0100
+++ cmd/kubeadm/app/util/config/testdata//defaulting/master/defaulted_darwin.yaml	2019-02-23 16:38:52.000000000 +0100
@@ -76,6 +76,10 @@
 portRange: ""
 resourceContainer: /kube-proxy
 udpIdleTimeout: 250ms
+winkernel:
+  enableDSR: false
+  networkName: ""
+  sourceVip: ""
 ---
 address: 0.0.0.0
 apiVersion: kubelet.config.k8s.io/v1beta1
```

Reply:
```
+winkernel:
+  enableDSR: false
+  networkName: ""
+  sourceVip: ""
```

that is because these fields need to be added to `defaulted_non_linux.yaml` too.
original PR that added the flags only updated our `defaulted.yaml`.

i.e. these fields are present on all platforms.

Reply:
Wouldn't that break this test on windows?

Reply:
Let me elaborate on this a bit. I'm just guessing here as I don't have windows machine to prove this, but looking at this PR: https://github.com/kubernetes/kubernetes/pull/70896/commits/164f79e2d4321dbf00c56345b20b015e856a497e I'd say it would add winkernel to non_linux testdata if it works on windows.

Reply:
i will close this PR and send one later today for you to test on OSX.
thank you!

/close

Reply:
I don't understand this to be honest. Why to close it if it fixes tests on MacOS?

Reply:
i will send a PR that tries to fix all OS and ping you on it.

Reply:
this is exactly what this PR already does.

Reply:
I'd understand closing it if you prove that it breaks tests on other platforms. But even with that I'd just fix the issues.